nala/features/personalization/mep-actions.test.js

```diff
@@ -0,0 +1,91 @@
+// to run tests:
+// npm run nala stage tag=mepact1 mode=headed
+
+import { expect, test } from '@playwright/test';
+import { features } from './mep-actions.spec.js';
+import TextBlock from '../../blocks/text/text.page.js';
+import MarqueeBlock from '../../blocks/marquee/marquee.page.js';
+
+const miloLibs = process.env.MILO_LIBS || '';
+
+// Test 0: confirm the default page
+test(`[Test Id - ${features[0].tcid}] ${features[0].name},${features[0].tags}`, async ({ page, baseURL }) => {
+  const textBlock1 = new TextBlock(page, 1);
+  const textBlock7 = new TextBlock(page, 7);
+  const URL = `${baseURL}${features[0].path}${miloLibs}`;
+  console.info(`[Test Page]: ${URL}`);
+  await page.goto(URL);
+  await expect(textBlock1.headlineAlt).toHaveText('Base page text. Section 2');
+  await expect(textBlock7.headlineAlt).toHaveText('Base page text fragment');
+});
+
+// Test 1: confirm various MEP actions on the personalized page
+test(`[Test Id - ${features[1].tcid}] ${features[1].name},${features[1].tags}`, async ({ page, baseURL }) => {
+  const textBlock1 = new TextBlock(page, 1);
```

Comment:
```suggestion
// looks good but quite repetitive. You can optimize this code to be less wordy :) Something like this could work 
  const actions = [
    'Inserted after the marquee',
    'Inserted before section2 text',
    'Base page text. Section 2',
    'Base page text. Section 3',
    'Appended to 3',
    'Prepended to 4',
    'Base page text. Section 4',
    null,
    'Section 6 replacement',
    'Base page text. Section 7',
    'Replaced basepage fragment',
    'Inserted after basepage fragment',
    'Inserted after replaced fragment'
  ]

  for (let i = 0; i < actions.length; i++) {
    const textBlock = new TextBlock(page, i + 1);
    const action = actions[i];
    if(action)
      await expect(textBlock.headlineAlt).toHaveText(actions[i]);
    else
      await expect(textBlock.headlineAlt).not.toBeVisible();
  }
```